work_queue/src/bindings/python3/PyTask/PyTask.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+# copyright (C) 2021- The University of Notre Dame
+# This software is distributed under the GNU General Public License.
+# See the file COPYING for details.
+
+import work_queue as wq
+import dill
+import os
+import sys
+import tempfile
+import uuid
+import traceback
+import shutil
+import textwrap
+
+
+class PyTask(wq.Task):
+	
+	p_tasks = []
+
+	def __init__(self, func, *args):
+		
+		self._id = str(uuid.uuid4())
+		self._tmpdir = tempfile.mkdtemp()
+
+		self._func_file = os.path.join(self._tmpdir, 'function_{}.p'.format(self._id))
+		self._args_file = os.path.join(self._tmpdir, 'args_{}.p'.format(self._id))
+		self._out_file = os.path.join(self._tmpdir, 'out_{}.p'.format(self._id))
+		self._wrapper = os.path.join(self._tmpdir, 'pytask_wrapper.py'.format(self._id))
+		#self._package_file = os.path.join(self._tmpdir, 'dep.yml')
+		self._tar_file  = 'venv.tar.gz'
+		self._pp_run = shutil.which('python_package_run')		
+
+		self._command = self.python_function_command(func, *args)
+		self.create_wrapper()
+
+		super(PyTask, self).__init__(self._command)
+		
+		self.specify_IO_files()
+		
+
+		PyTask.p_tasks.append(self)
+		
+	
+	def python_function_command(self, func, *args):
+			
+	
+		tb = traceback.extract_stack()
+		tb_list = traceback.format_list(tb)
+		caller = tb_list[len(tb_list)-3].split()[1]
+		caller = caller.replace('"', '')
+		caller = caller.replace(',', '')
+
+
+	#	if not os.path.exists(tar_file):
+	#		os.system('python_package_analyze ' + caller + ' ' + package_file)
+
+	#	os.system('python_package_create x.yml ' + self._tar_file)
+			
+		
+		with open(self._func_file, 'wb') as wf:
+			dill.dump(func, wf)
+		
+		with open(self._args_file, 'wb') as wf:
+			dill.dump([*args], wf)
+		
+		#command = "python {wrapper} {function} {args} {out}".format(
+		command = "./{pprun} -e {tar} python {wrapper} {function} {args} {out}".format(
```

Comment:
Here lets do a:
```python
'./{pprun} -e {tar} --unpack-to "$WORK_QUEUE_SANDBOX"/{unpack}-env  python {wrapper} {function} {args} {out}".format(
...
unpack=os.path.basename(self._tar_file)
...
```

Reply:
what does this do?

Reply:
Instead of unpacking the environment for each task, it unpacks it once to a known location so that further tasks can reuse it. It saves about 20s per task (Other than the very first task executing in a worker.)